### PR TITLE
Show attachments in virkailija application view

### PR DESCRIPTION
**Pull request by elahti**

### CODE REVIEW

File: src/cljs/ataru/virkailija/views/virkailija_readonly.cljs [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+(defn row-container [application lang children]
+  (let [ui (subscribe [:state-query [:application :ui]])]
+    (fn [application lang children]
+      (into [:div] (child-fields children application lang @ui)))))
+
+(defn- extract-values [children answers]
+  (let [child-answers  (->> (map answer-key children)
+                            (select-keys answers))
+        ; applicant side stores values as hashmaps
+        applicant-side (map (comp
```
Comment (wolverian) [inline src/cljs/ataru/virkailija/views/virkailija_readonly.cljs]: Can we write this as:

```clojure
(map (comp 
      (map :value) 
      :values 
      second))
```

In other words, lift the explicit mapping over `values` into a transducer.

File: src/cljs/ataru/virkailija/application/handlers.cljs [hunk 2]
```diff
@@ -151,19 +154,73 @@
                                                                                         :score
                                                                                         :state])}))})))
 
-(reg-event-db
+(reg-event-fx
+  :application/handle-fetch-application-attachment-metadata
+  (fn [{:keys [db]} [_ response]]
+    (let [response-map (reduce (fn [db {:keys [key] :as metadata}]
```
Comment (wolverian) [inline src/cljs/ataru/virkailija/application/handlers.cljs]: Should we call the accumulator something else than `db` here?

File: .gitignore [hunk 3]
```diff
@@ -25,3 +25,4 @@ config/ui-test.edn
 .swp
 /logs
 .oracle_jre_usage
+npm-debug.log*
```
Comment (wolverian) [inline .gitignore]: 👍

File: src/clj/ataru/files/file_store.clj [hunk 4]
```diff
@@ -23,9 +23,19 @@
...
+(defn delete-file [file-key]
+  (let [url  (str (get-in config [:liiteri :url]) "/api/files/" file-key)
         resp @(http/delete url)]
-    (clojure.pprint/pprint resp)
+    (when (= (:status resp) 200)
+      (json/parse-string (:body resp) true))))
+
+(defn get-metadata [application-keys]
+  (let [query-part (->> application-keys
+                        (map-indexed (fn [idx key]
```
Comment (elahti) [inline src/clj/ataru/files/file_store.clj]: Yes 👍

File: src/clj/ataru/files/file_store.clj [hunk 5]
```diff
@@ -23,9 +23,19 @@
...
-(defn delete-file [key]
-  (let [url  (str (get-in config [:liiteri :url]) "/api/files/" key)
+(defn delete-file [file-key]
+  (let [url  (str (get-in config [:liiteri :url]) "/api/files/" file-key)
         resp @(http/delete url)]
-    (clojure.pprint/pprint resp)
+    (when (= (:status resp) 200)
+      (json/parse-string (:body resp) true))))
+
+(defn get-metadata [application-keys]
```
Comment (elahti) [inline src/clj/ataru/files/file_store.clj]: Yes 👍

File: src/clj/ataru/files/file_store.clj (hunk 4 above)
Comment (elahti) [inline src/clj/ataru/files/file_store.clj]: Same functionality can be found at ataru.virkailija.application.handlers.cljs:183

File: src/clj/ataru/virkailija/virkailija_routes.clj [hunk 6]
```diff
@@ -254,7 +255,17 @@
                                        :path-params [koodisto-uri :- s/Str version :- Long]
                                        :return s/Any
                                        (let [koodi-options (koodisto/get-koodisto-options koodisto-uri version)]
-                                         (ok koodi-options))))))
+                                         (ok koodi-options))))
+
+                 (api/context "/files" []
+                   :tags ["files-api"]
+                   (api/GET "/" []
+                     :query-params [key :- (api/describe [s/Str] "File key")]
```
Comment (elahti) [inline src/clj/ataru/virkailija/virkailija_routes.clj]: -> `keys`

File: src/cljs/ataru/hakija/hakija_readonly.cljs [hunk 7]
```diff
@@ -1,11 +1,4 @@
-; Note: the css classes used below have different css implementations
```
Comment (elahti) [inline src/cljs/ataru/hakija/hakija_readonly.cljs]: Restored

File: src/cljs/ataru/virkailija/application/handlers.cljs [hunk 8]
```diff
@@ -4,17 +4,20 @@
...
             [ataru.virkailija.application-sorting :as application-sorting]
             [reagent.core :as r]
-            [taoensso.timbre :refer-macros [spy debug]]))
+            [taoensso.timbre :refer-macros [spy debug]]
+            [ataru.feature-config :as fc]))
 
 (reg-event-fx
   :application/select-application
   (fn [{:keys [db]} [_ application-key]]
     (if (not= application-key (get-in db [:application :selected-key]))
```
Comment (elahti) [inline src/cljs/ataru/virkailija/application/handlers.cljs]: Apparently returning `nil` from a event handler does nothing. App state is preserved and re-frame says nothing. Let's call it a feature?

File: src/clj/ataru/virkailija/virkailija_routes.clj (hunk 6 above)
Comment (elahti) [inline src/clj/ataru/virkailija/virkailija_routes.clj]: No

**Pull request merged**